dvc/repo/metrics/show.py

```diff
@@ -269,6 +269,9 @@ def show(
         if metrics:
             res[branch] = metrics
 
+    if all_tags and not all_branches:
```

Comment:
Should the brancher itself return it though?

Reply:
As discussed in private:
It should, though, that might cause some problems. For example:
- we `run` some command, (dvc add it to cache)
- we don't push
- we run `dvc gc -T`
- during gc, only tags will be preserved
- in result we loose `run` results from cache

Reply:
@pared and how about returning the working tree, but not the current branch? As you've noted, we do return "Working tree" when is_dirty(), but maybe we should return "working tree" even if it is not dirty? Effectively when on `master`, it will return `master`. That way `gc` and other functions gather current state + what has been added with -T/-a

Reply:
Agreed, but then `metrics show -T` will still return working tree. We should consider whether #2079 is actually a bug.

Reply:
But #2079 is about `master` not `working tree`

Reply:
returning working tree seems reasonable in all applications

Reply:
Agreed, ill introduce appropriate changes